### PR TITLE
Added nav menu to site config

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
N/A

#### What's this PR do?
Adds a nav menu to the site config so that published course sites can have a Hugo menu configured

#### How should this be manually tested?
- Paste the site config into a new Website Starter in your local instance of `ocw-studio` (or on RC). The Django admin field is equipped to handle yaml or json, and will be automatically converted to json if yaml is pasted in.
- Create a Website using this new Website Starter, and create a few pages. 
- Go to settings > menu, add those pages and some external links to the menu, and save
- Check your exported site in Github and navigate to `/config/menu-sidenav.yml`. The content of that file should be a valid Hugo menu format, and should match the menu you arranged in OCWS

### CODE REVIEW

File: ocw-www/ocw-studio.yaml [hunk 1]
```diff
@@ -184,3 +184,17 @@ collections:
         name: headless
         widget: hidden
         default: true
+
+  - category: Settings
+    label: Menu
+    name: menu
+    files:
+      - file: config/menu-sidenav.yml
```
Comment (gumaerc) [inline ocw-www/ocw-studio.yaml:192]: Unfortunately, in order for this to work with Hugo all menus must be in a file called `menus.yaml`.  More info on that here:

https://gohugo.io/getting-started/configuration/#configuration-directory

You might notice that you can create separate menu config files for different languages, but it's really designed only to be used for different language codes.  I've experimented with this and if you create a menu file that doesn't match a configured language then Hugo will throw an error at build time.  For instance, I tried creating a `menus.leftnav.yaml` file and then Hugo complained that a menu for `en` (the configured language code) doesn't exist.  So we will have to use the one `menus.yaml` file, then the way we separate different menus are with the root key in this file.  Here is what was created after I tested this:

```
menu:
  mainmenu:
  - identifier: bc941918-fe95-4c4a-8269-87a56bb93076
    name: Test Document
    url: content/resources/test-document.md
    weight: 0
 ```

In the `menus.yaml` file, the `menu` key is not necessary.  This is how it should look:

```
mainmenu:
- identifier: bc941918-fe95-4c4a-8269-87a56bb93076
  name: Test Document
  url: content/resources/test-document.md
  weight: 0
```

**Review by gumaerc — CHANGES_REQUESTED** [review]
I have some concerns about Hugo specific things with this implementation:

File: ocw-www/ocw-studio.yaml (hunk 1 above)
Comment (gumaerc) [reply ocw-www/ocw-studio.yaml:192]: To clarify, the file these need to go in should be stored at `config/_default/menus.yaml`.  I made an issue for the root level key problem in `ocw-studio` here: https://github.com/mitodl/ocw-studio/issues/416

**Comment by gsidebo** [comment]
@gumaerc The root issue has been fixed (https://github.com/mitodl/ocw-studio/pull/425), and I updated the config in this PR

**Review by gumaerc — APPROVED** [review]
:+1:

**Pull request merged**